src/Subscription.js

```diff
@@ -327,14 +238,7 @@ class Subscription extends EventEmitter {
          * If parsing the (expected) message failed, we should still mark it as received. Otherwise the
          * gap detection will think a message was lost, and re-request the failing message.
          */
-        let key
-        if (err.streamMessage) {
-            key = err.streamMessage.getPublisherId() + err.streamMessage.messageId.msgChainId
-        }
-
-        if (err instanceof Errors.InvalidJsonError && !this.checkForGap(err.streamMessage.prevMsgRef, key)) {
-            this.lastReceivedMsgRef[key] = err.streamMessage.getMessageRef()
-        }
+        this.orderingUtil.addError(err)
```

Comment:
Related to the `addError` comments above. Move the digging of streamMessage from error maybe here?